src/module-elasticsuite-catalog/Model/Product/Indexer/Fulltext/Datasource/PriceData.php

```diff
@@ -65,10 +76,26 @@ public function addData($storeId, array $indexData)
             $originalPrice = $priceModifier->getOriginalPrice($priceDataRow);
             $price         = $priceModifier->getPrice($priceDataRow);
 
+            $isDiscount    = $price < $originalPrice;
+            if (in_array($productTypeId, $this->attributeResourceModel->getCompositeTypes())) {
+                $isDiscount = false;
+                $priceModifier = $this->getPriceDataReader('default');
+                $allChildrenIds = $this->attributeResourceModel->loadChildrens([$productId], $storeId);
+                $childPriceData = $this->resourceModel->loadPriceData($storeId, array_keys($allChildrenIds));
```

Comment:
same as above, you are doing a query for each "batch" of chil products, that could probably be done with an unique query pre-emptively.